.gitlab-ci.yml

```diff
@@ -144,6 +144,55 @@ build-review-image:
       --build-arg "ARG_CI_COMMIT_BRANCH=${CI_COMMIT_BRANCH}"
       --build-arg "ARG_CI_COMMIT_SHA=${CI_COMMIT_SHA}"
 
+build-idp-image:
+  stage: review
+  needs: []
+  interruptible: true
+  variables:
+    BRANCH_TAGGING_STRING: ''
+  rules:
+    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+      variables:
+        BRANCH_TAGGING_STRING: '--destination ${ECR_REGISTRY}/identity-idp/idp:main'
+    - if: $CI_COMMIT_BRANCH != $CI_DEFAULT_BRANCH
+    - if: $CI_PIPELINE_SOURCE != "merge_request_event"
+      when: never
+  tags:
+    - build-pool
+  image:
+    name: gcr.io/kaniko-project/executor:debug
+    entrypoint: ['']
+  script:
```

Comment:
Would it make sense to move some of these more involved CI scripts out of YAML and into dedicated script files that are then invoked by the YAML? That way we can maybe take more advantage of linters / etc for the scripts and keep the yaml more for orchestration and managing the relationships between jobs?

For example, for this one, we could create a new script in i.e. `/scripts/ci/build-idp-image.sh`, and have the `script:` element of this job just invoke that script.

Reply:
To be clear: do not let this block your work, just something that's been on my mind!

Reply:
Maybe?  It's something that I wonder about too, because we have some really big inline scripts in identity-devops that are kinda ugly.  But at the same time, one of the things I'm trying to do is remove all the layers of scripts and abstractions that we have on top our infrastructure code so that it's not as hard to understand and change things.  I don't really know the right balance yet.

If you really think that you would understand it better if it were a separate script, let me know, and I'll refactor.

Reply:
I can't tell if my opinions are just because I don't work much with gitlab ci / k8s in my day-to-day. I'll defer to y'all!

Reply:
I generally agree, but I don't have a hard line. I think if it's a script that we'd want to run locally outside of CI for testing or it's complex enough (multiple commands and/or some logic involved), it should probably be a separate script.